WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -438,6 +449,12 @@ private void buildAndShowNotificationFromNoteData(Context context, Bundle data)
             showGroupNotificationForBuilder(context, builder, wpcomNoteID, message);
         }
 
+        private void showSimpleNotification(Context context, String title, String message, Intent resultIntent,
+                                            int pushId) {
+            NotificationCompat.Builder builder = getNotificationBuilder(context, title, message);
```

Comment:
here, note that `getNotificationBuilder` internally creates a Builder on the `notification_channel_normal_id` notification channel, while we may actually want Zendesk notifications to be on our `notification_channel_important_id` channel. Maybe, we might want to expose that as a parameter of `getNotificationBuilder` and refactor accordingly, or maybe just duplicate the method and make it specific, like `getNotificationBuilderForZendesk()` and `showSimpletNotificationForZendesk()`, etc.

Reply:
I'll have a look if we decide to go with the important channel. There is some ... _unfortunate_ code `GCMMessageService`. I might have a look at these next week during the hack week.

Just to give an idea of what this method could have been, here are a few things we could do:

* We shouldn't have 3 instances `new NotificationCompat.Builder` and use a single method with high(er) customization.
* All notification builders has to have a channel id and we have 2 of them (for now). That means we should have had it in the parameters from the start.
* To expand on that ^, ideally, the parameter would not be a string but an enum. Although a boolean could have been enough for now (since we have 2 values), an enum would have given us an easy way to build on it. The enum could have the strings in it, so it's all very clear code and very nice code completion.
* For bonus points, if we can switch to Kotlin when we refactor this class, we should use named parameters with default values, so it has maximum customizability with very little code. More importantly, there is a lot of things a method that returns a builder could be doing and there is little transparency about it if you don't look at the source code. The best thing about named parameters is that just by looking at code completion you can tell what the code does. (well.. 🤞 )

Anyway, I felt that just saying _unfortunate_ wouldn't be very constructive, so I wanted to drop some ideas for this simple case. If we want to do some actual improvement in this class, we should have a look at methods like `showGroupNotificationForBuilder`, specifically how it's getting a builder and then completely overriding it and using a different instance. That type of stuff is what I am trying not to 👀 at in order to avoid stabbing my eyes. 😭